homeassistant/components/flux_led/light.py

```diff
@@ -296,18 +297,25 @@ def _turn_on(self, **kwargs):
             return
         # Preserve current brightness on color/white level change
         if hs_color is not None:
-            if brightness is None:
-                brightness = self.brightness
-            color = (hs_color[0], hs_color[1], brightness / 255 * 100)
+            # Special case warm-white mode
+            if hs_color == (29.6, 58.824):
+                white = 255 if brightness is None else self.brightness
```

Comment:
in "warm white" mode, `white` value essentially becomes `brightness`. If `brightness` is None, it means that the lights coming from off to on state. Therefore, the best possible value `white` should be set to is 255 (max value)